components/ai_chat/core/browser/engine/engine_consumer_oai.cc

```diff
@@ -0,0 +1,234 @@
+}
+
+void EngineConsumerOAIRemote::GenerateQuestionSuggestions(
+    const bool& is_video,
+    const std::string& page_content,
+    SuggestedQuestionsCallback callback) {
+  const std::string& truncated_page_content =
+      page_content.substr(0, max_page_content_length_);
+  std::string content_segment = base::ReplaceStringPlaceholders(
+      l10n_util::GetStringUTF8(is_video
+                                   ? IDS_AI_CHAT_CLAUDE_VIDEO_PROMPT_SEGMENT
+                                   : IDS_AI_CHAT_CLAUDE_ARTICLE_PROMPT_SEGMENT),
+      {truncated_page_content}, nullptr);
+
+  base::Value::List messages;
+
+  {
+    base::Value::Dict message;
+    message.Set("role", "system");
+    message.Set("content",
+                "Separate each suggested question with the character |");
+    messages.Append(std::move(message));
+  }
+
+  {
+    base::Value::Dict message;
+    message.Set("role", "user");
+    message.Set("content", base::StrCat({content_segment, kQuestionPrompt}));
+    messages.Append(std::move(message));
+  }
+
+  api_->PerformRequest(
+      model_, std::move(messages), base::NullCallback(),
+      base::BindOnce(
+          &EngineConsumerOAIRemote::OnGenerateQuestionSuggestionsResponse,
+          weak_ptr_factory_.GetWeakPtr(), std::move(callback)));
+}
+
+void EngineConsumerOAIRemote::OnGenerateQuestionSuggestionsResponse(
+    SuggestedQuestionsCallback callback,
+    GenerationResult result) {
+  if (!result.has_value() || result->empty()) {
+    // Query resulted in error
+    LOG(ERROR) << "Error getting question suggestions.";
+    std::move(callback).Run(base::unexpected(std::move(result.error())));
+    return;
+  }
+
+  std::vector<std::string> questions =
+      base::SplitString(*result, "|", base::WhitespaceHandling::TRIM_WHITESPACE,
+                        base::SplitResult::SPLIT_WANT_NONEMPTY);
+  std::move(callback).Run(std::move(questions));
+}
+
+void EngineConsumerOAIRemote::GenerateAssistantResponse(
+    const bool& is_video,
+    const std::string& page_content,
+    const ConversationHistory& conversation_history,
+    const std::string& human_input,
+    GenerationDataCallback data_received_callback,
+    GenerationCompletedCallback completed_callback) {
+  if (conversation_history.empty()) {
+    std::move(completed_callback).Run(base::unexpected(mojom::APIError::None));
+    return;
+  }
+
+  const mojom::ConversationTurnPtr& last_turn = conversation_history.back();
+  if (last_turn->character_type != CharacterType::HUMAN) {
+    std::move(completed_callback).Run(base::unexpected(mojom::APIError::None));
+    return;
+  }
+
+  std::optional<std::string> selected_text = std::nullopt;
+  if (last_turn->selected_text.has_value()) {
+    selected_text =
+        last_turn->selected_text->substr(0, max_page_content_length_);
+  }
+  const std::string& truncated_page_content = page_content.substr(
+      0, selected_text ? max_page_content_length_ - selected_text->size()
```

Comment:
@LorenzoMinto `max_page_content_length_` what's an optimal char limit for custom models? for truncation

Reply:
see other comment.